python/hsfs/engine/python.py

```diff
@@ -933,17 +925,20 @@ def acked(err, msg):
         if offline_write_options is not None and offline_write_options.get(
             "start_offline_backfill", True
         ):
-            print("Launching offline feature group backfill job...")
-            self._job_api.launch(job_name)
-            print(
-                "Backfill Job started successfully, you can follow the progress at \n{}".format(
-                    self._get_job_url(job.href)
-                )
-            )
-            self._wait_for_job(job, offline_write_options)
+            self.run_job(job, offline_write_options)
 
         return job
 
+    def run_job(self, job, write_options=None):
+        print(f"Launching job: {job.name}")
+        self._job_api.launch(job.name)
+        print(
+            "Job started successfully, you can follow the progress at \n{}".format(
+                self._get_job_url(job.href)
+            )
+        )
+        self._wait_for_job(job, write_options)
```

Comment:
I would move this method to the Job class and replace the `write_options` with a simple parameter `await_termination` so that you don't need to leak the feature store specific write_options to the jobs api.

You might need to move the `JobsApi` class to the `Jobs` class, but I think that's ok.

Reply:
there is the following issue:
`tests/engine/test_python.py:None (tests/engine/test_python.py)
test_python.py:23: in <module>
    from hsfs import (
../../hsfs/__init__.py:20: in <module>
    from hsfs.connection import Connection
../../hsfs/connection.py:23: in <module>
    from hsfs import engine, client, util
../../hsfs/engine/__init__.py:17: in <module>
    from hsfs.engine import spark
../../hsfs/engine/spark.py:67: in <module>
    from hsfs import feature, training_dataset_feature, client, util
../../hsfs/training_dataset_feature.py:19: in <module>
    import hsfs.feature_group as feature_group
../../hsfs/feature_group.py:31: in <module>
    from hsfs.core import (
../../hsfs/core/feature_group_engine.py:21: in <module>
    from hsfs.core import feature_group_base_engine, hudi_engine
../../hsfs/core/feature_group_base_engine.py:17: in <module>
    from hsfs.core import feature_group_api, storage_connector_api, tags_api, kafka_api
../../hsfs/core/feature_group_api.py:19: in <module>
    from hsfs.core import ingestion_job
../../hsfs/core/ingestion_job.py:18: in <module>
    from hsfs.core.job import Job
../../hsfs/core/job.py:19: in <module>
    from hsfs.core import job_api
../../hsfs/core/job_api.py:21: in <module>
    class JobApi:
../../hsfs/core/job_api.py:39: in JobApi
    def get(self, name: str) -> job.Job:
E   AttributeError: partially initialized module 'hsfs.core.job' has no attribute 'Job' (most likely due to a circular import)
`